cpp/src/copying/copy.cu

```diff
@@ -334,6 +354,8 @@ std::unique_ptr<column> copy_if_else(column_view const& lhs,
   CUDF_EXPECTS(boolean_mask.size() == lhs.size(),
                "Boolean mask column must be the same size as lhs and rhs columns");
   CUDF_EXPECTS(lhs.size() == rhs.size(), "Both columns must be of the size");
+  CUDF_EXPECTS(lhs.type() == rhs.type(), "Both inputs must be of the same type");
```

Comment:
This seems inconsistent with the PR description:
```
 The column/column version will accept 2 dictionary columns with matching key types.
```

Reply:
The key type matching check is done inside the call to scatter. Here is where it is tested: https://github.com/rapidsai/cudf/pull/9887/files#diff-0ae8b9ed14e6b0f689e80f46339660b3dfff12a78e2279632bbd727fa1dfa529R713